### PR TITLE
Refactor dockerfiles to use multi-stage

### PR DESCRIPTION
**Pull request by jdmaguire**
We have a lot of dockerfiles, there are some issues:

- The name format is irregular. Normal Dockerfile name format is either `(<name>.)Dockerfile` or `Dockerfile(.<name>)` whereas we are doing `Dockerfile-<name>`. It isn't consistent even with other dockerfiles in this repo.
- We have a lot of dockerfiles where Dockerfile 2 relies on Dockerfile 1. Say for files. The normal way one would do this would be a multi-stage dockerfile. Instead we have multiple files and have bash doing the work of building one, then supplying the name to the next step.

This PR reorganizes the docker files. Seven dockerfiles into two dockerfiles. A few dozen lines of bash code removed by letting docker do more lifting.

This doesn't improve build speed but I feel makes it easier to read what is being built & will make it easier to later increase speed.

### CODE REVIEW

**Review by claudiofahey — CHANGES_REQUESTED** [review]
Thanks for cleaning this up. I just have some minor feedback.

File: pravega-video-server/src/main.rs [hunk 1]
```diff
@@ -40,7 +40,7 @@ fn main() {
     let config = ClientConfigBuilder::default()
         .controller_uri(controller_uri)
         .build()
-        .expect("creating config");
+        .expect("creating config ok");
```
[the remark this answers is not in the document]
Comment (jdmaguire) [reply pravega-video-server/src/main.rs:43]: Was testing something. Didn't mean to push.

**Comment by jdmaguire** [comment]
The push-release.sh needs some editing

**Comment by claudiofahey** [comment]
> The push-release.sh needs some editing

You can delete it. We don't use this file. Note I took most of the Docker files from https://github.com/restreamio/docker-gstreamer. That is why there are unused files.

File: docker/build-release.sh [hunk 2]
```diff
@@ -41,26 +36,15 @@ docker build -t pravega/gstreamer:dev-downloaded \
...
+    .
 popd
 
 # Build pravega-dev image which includes the source code and binaries for all applications.
 docker build -t pravega/gstreamer:pravega-dev \
     --build-arg FROM_IMAGE=pravega/gstreamer:${GSTREAMER_CHECKOUT}-dev-with-source \
+    --build-arg RUST_JOBS=${RUST_JOBS} \
     -f ${ROOT_DIR}/docker/pravega-dev.Dockerfile ${ROOT_DIR}
+
+#CARGO_TARGET_DIR environment /cache/build/rust
```
[the remark this answers is not in the document]
Comment (jdmaguire) [reply docker/build-release.sh:50]: Ignore. Accidentally committed ;) it is a supprise for another time.

[the remark this answers is not in the document]
Comment (jdmaguire) [reply docker/build-release.sh:50]: (Will remove for this PR.)

File: docker/build-latest.sh [hunk 3]
```diff
@@ -14,17 +14,10 @@ set -ex
```
[the remark this answers is not in the document]
Comment (jdmaguire) [reply docker/build-latest.sh:14]: The `push-release.sh` relies on the images built here I believe?

**Review by claudiofahey — APPROVED** [review]

**Pull request merged**